test/TestCriterion.cpp

```diff
@@ -36,6 +36,78 @@ TEST(TestCriterion, TestConstructorByCopy) {
   EXPECT_EQ(os.str(), "Criterion(id : test_copy, name : criterion_by_copy, direction : +, weight : 0.3)");
 }
 
+TEST(TestCriterion, TestGetterSetter) {
+  std::string id = "test";
+  std::string name = "criterion_test";
+  int direction = -1;
+  float weight = 0.4;
+  Criterion crit = Criterion(id, name, direction, weight);
+  EXPECT_EQ("test", crit.getId());
+  EXPECT_EQ("criterion_test", crit.getName());
+  EXPECT_EQ(-1, crit.getDirection());
+  EXPECT_FLOAT_EQ(0.4, crit.getWeight());
+
+  crit.setId("test_set");
+  crit.setName("criterion_test_set");
+  crit.setDirection(1);
+  crit.setWeight(0.3);
+
+  EXPECT_EQ("test_set", crit.getId());
+  EXPECT_EQ("criterion_test_set", crit.getName());
+  EXPECT_EQ(1, crit.getDirection());
+  EXPECT_FLOAT_EQ(0.3, crit.getWeight());
+  
+}
+
+TEST(TestCriterion, TestDirectionGenerator) {
+  std::string id = "test_direction";
+  Criterion crit = Criterion(id);
+  unsigned long int seed = time(0);
+
+  std::string id2 = "test_direction2";
+  Criterion crit2 = Criterion(id2);
+  unsigned long int seed2 = 40;
+  // generating a direction to compare to the one generated in generateDirection
+  // should be the same since we are using the same seed 
+  srand(seed);
+  int direction = 1;
+  if ( ((float) rand() / RAND_MAX) < 0.5){ direction = -1; }
+  crit.generateDirection(seed);
+
+  srand(seed2);
```

Comment:
It looks like you are testing twice the same feature here

Reply:
Right, I just wanted to test with different seeds but not really useful. Will remove